### PR TITLE
SB-21592 email notitification, systemsetting and otp generation api

### PR DESCRIPTION
**Pull request by sknirmalkar89**

### CODE REVIEW

**Comment by keshavprasadms** [comment]
Hello @sknirmalkar89,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://630.qa.docs.sunbird.org/api-documentation/ .<br/> **Note:** Only _api-documentation_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://630.qa.docs.sunbird.org/api-documentation/apis/notificationapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/otpapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/otpapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/systemsettingsapi// <br />  <br/> **Modified sections are -** <br />

**Comment by keshavprasadms** [comment]
Hello @sknirmalkar89,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://630.qa.docs.sunbird.org/api-documentation/ .<br/> **Note:** Only _api-documentation_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://630.qa.docs.sunbird.org/api-documentation/apis/notificationapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/otpapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/otpapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/systemsettingsapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/systemsettingsapi// <br />  <br/> **Modified sections are -** <br />

File: apis/notificationapi/notificationapi.yaml [hunk 1]
```diff
@@ -1,12 +1,16 @@
 openapi: 3.0.1
 info:
   title: Notification API(s)
-  description: 'The Email resources provides the notification mechanism for all the
-    users on Sunbird. The Resource notifies users officially about something on Sunbird. '
+  description: The Email resources provides the notification mechanism for all the
```
Comment (Sunitashelar) [inline apis/notificationapi/notificationapi.yaml:4]: @sknirmalkar89 The description of this microservice is not understandable. Could you please clarify? What is meant by 'something'? To write it differently, please answer the following questions about the Notifications microservice:
1. Why do you need this microservice?
2. What does the microservice do?
3. When can one use this microservice?

File: apis/notificationapi/notificationapi.yaml [hunk 2]
```diff
@@ -15,10 +19,15 @@ paths:
       tags:
       - Notification APIs
       summary: ' Notifiying Users'
-      description: |-
-        This API is associated with notifying a user on the Sunbird Platform.if request will contain mode attribute with value sms ,then user will be notify via sms.During sms notification you can pass list of phone number or list of phone number and userid both.
-           - The endpoint for **Notifying Users** is `/user/v1/notification/email`
-           - The fields marked with an asterisk (*) are mandatory. They cannot be null or empty.
+      description: "This API is associated with notifying a user on the Sunbird Platform.if\
```
Comment (Sunitashelar) [inline apis/notificationapi/notificationapi.yaml:22]: @sknirmalkar89 This API description is not clear at all. Please explain. Also:
1. please do not use forced line breaks to format the text.
2. Start a new sentence with a capital letter. 
3. Check spelling and grammar

File: apis/otpapi/otpapi.yaml [hunk 3]
```diff
@@ -0,0 +1,407 @@
+openapi: 3.0.1
+info:
+  title: OTP API(s)
+  description: 'The Email resources provides the notification mechanism for all the
```
Comment (Sunitashelar) [inline apis/otpapi/otpapi.yaml:4]: @sknirmalkar89 this microservice description is a copy-paste of the notification description. Please correct it based on the comments given for the previous API.

File: apis/otpapi/otpapi.yaml [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+                  resmsgid: 
+                  msgid: 12eb0ca45ce4c776fab994d120bedd41
+                  err: null
+                  status: success
+                  errmsg: null.
+                responseCode: OK
+                result: 
+                 response: "SUCCESS"
+        400:
+          description: BAD REQUEST. The "Notifying User" operation failed ! The possible
```
Comment (Sunitashelar) [inline apis/otpapi/otpapi.yaml:67]: @sknirmalkar89 This is a copy-paste of the notify API. Please correct.

File: apis/otpapi/otpapi.yaml [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+                  errmsg: Process failed,please try again later.
+                responseCode: SERVER_ERROR
+                result: {}
+  /otp/v1/verify:
+    post:
+      tags:
+      - OTP APIs
+      summary: Verify OTP
+      description: |-
+        This API is associated with verifying OTP to user on the Sunbird Platform.
```
Comment (Sunitashelar) [inline apis/otpapi/otpapi.yaml:112]: @sknirmalkar89 does the API verify the OTP 'to' the user or 'of' the user? meaning does this API send a message to the user saying 'OTP verified' or does the API verify if the OTP entered by the user is correct? Please clarify and correct accordingly.

File: apis/otpapi/otpapi.yaml [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+          The body refers to the format of the request.
+            - The body contains metadata about the notification to be sent.
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/verifyOtpRequest'
+        required: true
+      responses:
+        200:
+          description: OK ! Successful operation."Notifying User" operation was successfully
```
Comment (Sunitashelar) [inline apis/otpapi/otpapi.yaml:136]: @sknirmalkar89 copy-paste of 'Notify' API. Please check globally. Also, check for forced line breaks globally.

File: apis/otpapi/otpapi.yaml [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+                  msgid: 12eb0ca45ce4c776fab994d120bedd41
+                  err: INTERNAL_ERROR
+                  status: INTERNAL_ERROR
+                  errmsg: Process failed,please try again later.
+                responseCode: SERVER_ERROR
+                result: {}
+components:
+  schemas:
+    emailRequest:
+      title: Request structure for sending email notification
```
Comment (Sunitashelar) [inline apis/otpapi/otpapi.yaml:186]: @sknirmalkar89 this entire request structure is for the email notifications, whereas the API is OTP.

File: apis/otpapi/otpapi.yaml [hunk 8]
```diff
@@ -0,0 +1,407 @@
...
+          type: object
+          properties: {}
+          description: 'API response code '
+    SendNotificationMailResult:
+      title: Send Notification Result
+      type: object
+      properties:
+        response:
+          type: string
+          description: SUCCESS in case mail send successfully otherwise user id with
```
Comment (Sunitashelar) [inline apis/otpapi/otpapi.yaml:364]: @sknirmalkar89 please check grammar.

**Review by Sunitashelar — CHANGES_REQUESTED** [review]

File: apis/notificationapi/notificationapi.yaml (hunk 1 above)
Comment (sknirmalkar89) [reply apis/notificationapi/notificationapi.yaml:4]: updated.

File: apis/otpapi/otpapi.yaml (hunk 3 above)
Comment (sknirmalkar89) [reply apis/otpapi/otpapi.yaml:4]: updated

File: apis/otpapi/otpapi.yaml (hunk 4 above)
Comment (sknirmalkar89) [reply apis/otpapi/otpapi.yaml:67]: updated

File: apis/otpapi/otpapi.yaml (hunk 5 above)
Comment (sknirmalkar89) [reply apis/otpapi/otpapi.yaml:112]: updated

File: apis/otpapi/otpapi.yaml (hunk 7 above)
Comment (sknirmalkar89) [reply apis/otpapi/otpapi.yaml:186]: updated

File: apis/otpapi/otpapi.yaml (hunk 8 above)
Comment (sknirmalkar89) [reply apis/otpapi/otpapi.yaml:364]: updated

File: apis/notificationapi/notificationapi.yaml (hunk 2 above)
Comment (sknirmalkar89) [reply apis/notificationapi/notificationapi.yaml:22]: updated

**Comment by keshavprasadms** [comment]
Hello @sknirmalkar89,<br/><br/> _This is a auto-generated response._ <br/><br/> Click the URL for review pull request - http://630.qa.docs.sunbird.org/api-documentation/ .<br/> **Note:** Only _api-documentation_ version was generated for PR preview. <br/><br/> **Modified pages are -** <br />- http://630.qa.docs.sunbird.org/api-documentation/apis/notificationapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/otpapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/otpapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/systemsettingsapi// <br /> - http://630.qa.docs.sunbird.org/api-documentation/apis/systemsettingsapi// <br />  <br/> **Modified sections are -** <br />

**Review by Sunitashelar — APPROVED** [review]

**Pull request merged**